### PR TITLE
Use URLSearchParams when parsing callback querystring

### LINKED ISSUE #1060: Invalid State error with Nuxt, custom callback page and url-encoded characters
### Describe the problem

I tried to implement a custom callback URL to allow for adjusted error handling in a Nuxt.js SPA.
Long story short: Nuxt.js seams to normalize the URL and to replace `%3D` by `=` in `location.href`.

Using the Vue.js integration works when using `createAuth0` as `handleRedirectCallback` is called _before_ that normalization.
When calling `handleRedirectCallback` from inside a Vue.js component that normalization has happened and the replacement causes the state from the transaction to not match that from the URL.

I believe this to be an error in auth0-spa-js due to the way it reads the URL.
The `Invalid State` error should happen whenever `=` characters are encoded in the state and those characters are replaced by anything before `handleRedirectCallback` is called.

**What I found out about the auth0 implementation:**

The `Invalid State` error is thrown in https://github.com/auth0/auth0-spa-js/blob/master/src/Auth0Client.ts#L513.
Key and Value of the URL are parsed by using `.split('=')` in https://github.com/auth0/auth0-spa-js/blob/master/src/utils.ts#L24.

💡 Reading the value by `new URL(location.href).searchParams.get('state')` gives the full value including trailing `'=='`.

`=` chars in the value should be very common, as the values are just base64 encoded random strings ( https://github.com/auth0/auth0-spa-js/blob/master/src/Auth0Client.ts#L293 ).

### What was the expected behavior?

The auth0 SDK should not fail if any middleware replaces query parameters by their URL-Decoded representation. This replacement might be done for human readability.

### Reproduction

- Try to Login via the Auth0 SDK
- Replace `%3D` with `=` in the URL before calling `handleRedirectCallback`
- call `handleRedirectCallback`

**Can the behavior be reproduced using the [SPA SDK Playground](https://github.com/auth0/auth0-spa-js/blob/master/DEVELOPMENT.md#the-sdk-playground)?**

### Environment

- **Version of `auth0-spa-js` used:** 1.22.5 (but behaviour shouldn't be changed as the respective code stayed the same) 
- **Which browsers have you tested in?**  Chrome
- **Which framework are you using, if applicable (Angular, React, etc):** Vue.js with Nuxt 3
- **Other modules/plugins/libraries that might be involved:** @auth0/auth0-vue

### PR DESCRIPTION
**Pull request by frederikprijck**
### Changes

Ensures we use URLSearchParams when parsing the QueryString while handling the redirect callback.
This was done differently in v1 because of IE support, but as we dropped IE support in v2 we can replace this with URLSearchParams.
 
### References
Fixes #1060 

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [x] All code quality tools/guidelines have been run/followed

### CODE REVIEW

File: __tests__/Auth0Client/helpers.ts [hunk 1]
```diff
@@ -42,7 +42,12 @@ export const assertPostFn = (mockFetch: jest.Mock) => {
     expect(url).toEqual(actualUrl);
 
     expect(body).toEqual(
-      json ? JSON.parse(call.body) : utils.parseQueryResult(call.body)
+      json
+        ? JSON.parse(call.body)
+        : Array.from(new URLSearchParams(call.body).entries()).reduce(
+            (acc, curr) => ({ ...acc, [curr[0]]: curr[1] }),
+            {}
+          )
```
Comment (frederikprijck) [inline __tests__/Auth0Client/helpers.ts:50]: This was (ab)using the `parseQueryResult` method, while it doesnt have anything to do with the redirectCallback.

File: __tests__/utils.test.ts [hunk 2]
```diff
@@ -23,29 +23,23 @@ afterEach(() => {
...
+  describe('parseAuthenticationResult', () => {
     it('parses the query string', () => {
       expect(
-        parseQueryResult('value=test&otherValue=another-test')
+        parseAuthenticationResult('code=test&state=another-test')
       ).toMatchObject({
-        value: 'test',
-        otherValue: 'another-test'
+        code: 'test',
+        state: 'another-test'
```
Comment (frederikprijck) [inline __tests__/utils.test.ts:32]: The type returned by `parseQueryResult`, only ever accounted for `code`, `state`, `error`, or `error_description`. This was testing a scenario that should not come up. So I changed it to code and state.

Probably an older test from when this code was used to parse any querystring.

File: src/utils.ts [hunk 3]
```diff
@@ -12,24 +12,21 @@ import {
   PopupCancelledError
 } from './errors';
 
-export const parseQueryResult = (queryString: string): AuthenticationResult => {
+export const parseAuthenticationResult = (
```
Comment (frederikprijck) [inline src/utils.ts:15]: Renamed this to `parseAuthenticationResult`, as it's not a generic querystring parser.

File: src/utils.ts [hunk 4]
```diff
@@ -12,24 +12,21 @@ import {
...
+  queryString: string
+): AuthenticationResult => {
   if (queryString.indexOf('#') > -1) {
-    queryString = queryString.substr(0, queryString.indexOf('#'));
+    queryString = queryString.substring(0, queryString.indexOf('#'));
```
Comment (frederikprijck) [inline src/utils.ts:19]: replaced deprecated `substr` with `substring`. Arguments are identical when you start at `0`.

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**